climate_scanner/coarse_article_classifier/data_utils.py

```diff
@@ -13,11 +21,31 @@ def pre_processing(self, text):
         Arguments: Str - Free Text Ex: Title / Document / Article etc.
         :return: Str - Cleaned text
         """
+
         text = text.lower()
+        # remove hyperlinks 
+        text = text.apply(lambda x: re.sub(r"http\S|www\.\S++", " ", x)) 
+        # remove hashtags
+        text = text.apply(lambda x: re.sub(r"#\w+", " ", x))
+        # remove html_tags
+        text = text.apply(lambda x: re.sub(r"<.*?>", " ", x))
+        # remove numbers
+        text = text.apply(lambda x: re.sub(r"\d+", " ", x))
+        # encode unknown
+        text = text.apply(lambda x: unicodedata.normalize("NFD", x).encode('ascii', 'ignore').decode("utf-8"))
+        # clean end-of-line tabs
+        text = text.str.lower()
+        text = text.apply(lambda x: x.replace("\n", " "))
+        text = text.apply(lambda x: x.replace("\r", " "))
+        text = text.apply(lambda x: x.replace("\t", " "))
+        # clean punctuations
+        text = text.apply(lambda x: re.sub(r'[^\w\s]', ' ', x))
+        # remove stopwords
+        text = text.apply(lambda x: self._remove_stop_words(x))
```

Comment:
@sonaalll  - when doing contextual embeddings especially, we shouldn't be removing stop words - we might lose some important tokens which will create a completely different embedding. We can remove extra spaces, HTML, Unicode characters. I am also thinking about the numbers - @Eliotfrazier  - I think we can keep them as well! Let us know your thoughts!